### PR TITLE
OpenMP backend refactor files.

### PR DESCRIPTION
**Pull request by rgayatri23**
The PR attempts to refactor the OpenMP backend implementation into files specific to each parallel patterns.

The PR does not change anything related to implementation.

### CODE REVIEW

File: core/src/OpenMP/Kokkos_OpenMP.hpp [hunk 1]
```diff
@@ -235,6 +235,9 @@ struct MemorySpaceAccess<Kokkos::OpenMP::memory_space,
 #include <OpenMP/Kokkos_OpenMP_Instance.hpp>
 #include <OpenMP/Kokkos_OpenMP_Team.hpp>
 #include <OpenMP/Kokkos_OpenMP_Parallel.hpp>
+#include <OpenMP/Kokkos_OpenMP_Parallel_For.hpp>
+#include <OpenMP/Kokkos_OpenMP_Parallel_Reduce.hpp>
+#include <OpenMP/Kokkos_OpenMP_Parallel_Scan.hpp>
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP.hpp:240]: No these probably should be included in `core/src/decl/Kokkos_Declare_OPENMP.hpp` instead.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp [hunk 2]
```diff
@@ -49,1219 +33,8 @@ inline bool execute_in_serial(OpenMP const& space = OpenMP()) {
           !(omp_get_nested() && (omp_get_level() == 1)));
 }
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:34]: If that is all that is left in `<okkos_OpenMP_Parallel.hpp>` then that file needs a new name that reflects better what it provides.  An alternative (better?) would be to move this into `<Kokkos_OpenMP.hpp>` or whatnot.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_Scan.hpp [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENMP_PARALLEL_REDUCE_SCAN_HPP
+#define KOKKOS_OPENMP_PARALLEL_REDUCE_SCAN_HPP
```
[the remark this answers is not in the document]
Comment (rgayatri23) [reply core/src/OpenMP/Kokkos_OpenMP_Parallel_Scan.hpp]: Duh. Sorry dint notice that. fixed it.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp (hunk 2 above)
Comment (rgayatri23) [reply core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp]: Moved this to `Kokkos_OpenMP_Instance.hpp`.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENMP_PARALLEL_FOR_HPP
+#define KOKKOS_OPENMP_PARALLEL_FOR_HPP
+
+#include <Kokkos_Macros.hpp>
+#if defined(KOKKOS_ENABLE_OPENMP)
```
[the remark this answers is not in the document]
Comment (rgayatri23) [reply core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp]: Ok removed that macro from all files. Hope all CI pass.

[the remark this answers is not in the document]
Comment (dalg24) [reply core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp]: I only suggested to drop these in the files that are being created now which as far as I can tell will compile just fine.
I am not sure where the error was coming from but the right fix will be to look where it was included from and guard there, not reverting to guarding in every single implementation file.

[the remark this answers is not in the document]
Comment (rgayatri23) [reply core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp]: Ah I misunderstood. I removed the macro for the files that are being changed.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_Scan.hpp [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENMP_PARALLEL_SCAN_HPP
+#define KOKKOS_OPENMP_PARALLEL_SCAN_HPP
+
+#include <Kokkos_Macros.hpp>
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel_Scan.hpp:20]:
```suggestion
```

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENMP_PARALLEL_FOR_HPP
+#define KOKKOS_OPENMP_PARALLEL_FOR_HPP
+
+#include <Kokkos_Macros.hpp>
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel_For.hpp:20]:
```suggestion
```

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_Reduce.hpp [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENMP_PARALLEL_REDUCE_HPP
+#define KOKKOS_OPENMP_PARALLEL_REDUCE_HPP
+
+#include <Kokkos_Macros.hpp>
```
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel_Reduce.hpp:20]:
```suggestion
```

**Review by dalg24 — APPROVED** [review]

File: core/src/OpenMP/Kokkos_OpenMP_Parallel_Reduce.hpp [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+                                        Kokkos::HostSpace>::accessible,
+        "Kokkos::OpenMP reduce result must be a View accessible from "
+        "HostSpace");
+  }
+};
+
+}  // namespace Impl
+}  // namespace Kokkos
+
+#endif /* KOKKOS_OPENMP_PARALLEL_REDUCE__HPP */
```
Comment (Rombur) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel_Reduce.hpp:567]:
```suggestion
#endif /* KOKKOS_OPENMP_PARALLEL_REDUCE_HPP */
```

**Review by Rombur — APPROVED** [review]

**Pull request merged**